apps/rush-lib/src/cli/actions/RebuildAction.ts

```diff
@@ -1,293 +0,0 @@
-  private _toFlag: CommandLineStringListParameter;
-  private _vsoParameter: CommandLineFlagParameter;
-  private _minimalParameter: CommandLineFlagParameter;
-  private _verboseParameter: CommandLineFlagParameter;
-
-  constructor(parser: RushCommandLineParser, options?: ICommandLineActionOptions) {
-    super(options || {
-      actionVerb: 'rebuild',
-      summary: 'Clean and rebuild the entire set of projects',
-      documentation: 'The Rush rebuild command assumes that the package.json file for each'
-      + ' project contains scripts for "npm run clean" and "npm run test".  It invokes'
-      + ' these commands to build each project.  Projects are built in parallel where'
-      + ' possible, but always respecting the dependency graph for locally linked projects.'
-      + ' The number of simultaneous processes will be equal to the number of machine cores.'
-      + ' unless overridden by the --parallelism flag.'
-    });
-    this._parser = parser;
-    this._isIncrementalBuildAllowed = false;
-  }
-
-  protected onDefineParameters(): void {
-    this._productionParameter = this.defineFlagParameter({
-      parameterLongName: '--production',
-      description: 'Perform a production build'
-    });
-    this._vsoParameter = this.defineFlagParameter({
-      parameterLongName: '--vso',
-      description: 'Display error messages in the format expected by the Visual Studio Team Services build interface'
-    });
-    this._npmParameter = this.defineFlagParameter({
-      parameterLongName: '--npm',
-      description: 'Perform a npm-mode build. Designed for building code for distribution on NPM'
-    });
-    this._parallelismParameter = this.defineIntegerParameter({
-      parameterLongName: '--parallelism',
-      parameterShortName: '-p',
-      key: 'COUNT',
-      description: 'Change limit the number of simultaneous builds. This value defaults to the number of CPU cores'
-    });
-    this._toFlag = this.defineStringListParameter({
-      parameterLongName: '--to',
-      parameterShortName: '-t',
-      key: 'PROJECT1',
-      description: 'Build the specified project and all of its dependencies'
-    });
-    this._fromFlag = this.defineStringListParameter({
-      parameterLongName: '--from',
-      parameterShortName: '-f',
-      key: 'PROJECT2',
-      description: 'Build all projects that directly or indirectly depend on the specified project'
-    });
-    this._minimalParameter = this.defineFlagParameter({
-      parameterLongName: '--minimal',
-      parameterShortName: '-m',
-      description: 'Invokes the build script with the "--minimal" option, which speeds up the build by running the ' +
-        'minimal set of tasks required to produce executable output'
-    });
-    this._verboseParameter = this.defineFlagParameter({
-      parameterLongName: '--verbose',
-      parameterShortName: '-v',
-      description: 'Display the logs during the build, rather than just displaying the build status summary'
```

Comment:
If all these flags are deleted, what happens to someone who currently relies on this flags in their repo? When they upgrade to the new Rush, will the flags all be mysteriously missing?

Maybe we could include the legacy settings in an "examples" folder, so people can simply copy it to their common/config/rush folder when they migrate?

Reply:
Yes, all the flags will be missing

---
In reply to: [146734888](https://github.com/Microsoft/web-build-tools/pull/387#discussion_r146734888) [](ancestors = 146734888)